### ISSUE TITLE
Assessment of the difficulty in porting CPU architecture for openal

### ISSUE THREAD

**Issue by wangyuliu** [kcat/openal-soft#913]
Hello everyone! I am working on implementing a tool to assess the complexity of CPU architecture porting. It primarily focuses on RISC-V architecture porting. In fact, it can be estimated on average across various architectures.My focus is on the overall workload and difficulty of transplantation in the past and future.As part of my dataset, I have collected the **openal-soft** project. **I would like to gather community opinions to support my assessment. I appreciate your help and response!** Based on scanning tools, the porting complexity is determined to be simple, with a small amount of code related to the CPU architecture in the project. (referring to the overall workload from adapting the project to a specific architecture to achieving full functionality on that architecture). Is this assessment accurate? I look forward to your help and response.

**Comment by kcat** [comment]
Yes, it should be relatively easy to port to other CPU architectures. As long as it uses 2's compliment signed integers and IEEE-754-like floating-point, it should only require a proper C++17 compiler and a good libc and STL, then it should already work in most cases. There are some CPU-specific intrinsics and inline assembly, primarily for optimized SIMD methods and float manipulation, but these are optional and there are generic fallbacks.

Performance may be somewhat compromised without optimized SIMD methods, depending on the compiler's ability to optimize and vectorize for the given target architecture, and there may need to be some consideration for denormals depending on if/how the architecture handles them (if they exist, or whether they can be disabled if they're a performance drain like on some x86 systems).

It wouldn't be hard to add optimized support with help from someone familiar with the architecture, but it should still be alright without it.
